blivet/partitioning.py

```diff
@@ -439,7 +441,12 @@ def getFreeRegions(disks):
     free = []
     for disk in disks:
         for f in disk.format.partedDisk.getFreeSpaceRegions():
-            if f.length >= disk.format.alignment.grainSize:
+            grain_size = disk.format.alignment.grainSize
+            if f.length >= grain_size:
+                if align:
+                    log.debug("lenght of free region aligned from %d to %d",
+                              f.length, f.length - (f.length % grain_size))
+                    f.length -= (f.length % grain_size)
```

Comment:
I'd calculate the new length once, just to make sure future changes don't miss one of them.